orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/OrcaMessageHandler.kt

```diff
@@ -89,15 +90,16 @@ internal interface OrcaMessageHandler<M : Message> : MessageHandler<M> {
   }
 
   fun Execution.shouldQueue(): Boolean {
-    if (!isLimitConcurrent) {
-      return false
+    val configId = pipelineConfigId
+    return when {
+      !isLimitConcurrent -> false
+      configId == null   -> false
+      else               -> {
+        val criteria = ExecutionCriteria().setLimit(2).setStatuses(RUNNING)
+        repository
+          .retrievePipelinesForPipelineConfigId(configId, criteria)
+          .any { it.id != id }
+      }
```

Comment:
I tried to make this code a little clearer but basically the logic is the same as before _except that_ it will return `false` if the only pipeline with the same config id that's running is the one you're asking about.